staging/src/k8s.io/kubectl/pkg/cmd/expose/expose.go

```diff
@@ -374,184 +370,26 @@ func (o *ExposeServiceOptions) RunExpose(cmd *cobra.Command, args []string) erro
-		var targetPort intstr.IntOrString
-		if portNum, err := strconv.Atoi(targetPortString); err != nil {
-			targetPort = intstr.FromString(targetPortString)
-		} else {
-			targetPort = intstr.FromInt(portNum)
-		}
-		// Use the same target-port for every port
-		for i := range service.Spec.Ports {
-			service.Spec.Ports[i].TargetPort = targetPort
-		}
-	} else {
-		// If --target-port or --container-port haven't been specified, this
-		// should be the same as Port
-		for i := range service.Spec.Ports {
-			port := service.Spec.Ports[i].Port
-			service.Spec.Ports[i].TargetPort = intstr.FromInt(int(port))
-		}
-	}
-	if len(o.ExternalIP) > 0 {
-		service.Spec.ExternalIPs = []string{o.ExternalIP}
-	}
-	if len(o.Type) != 0 {
-		service.Spec.Type = corev1.ServiceType(o.Type)
-	}
-	if service.Spec.Type == corev1.ServiceTypeLoadBalancer {
-		service.Spec.LoadBalancerIP = o.LoadBalancerIP
-	}
-	if len(o.SessionAffinity) != 0 {
-		switch corev1.ServiceAffinity(o.SessionAffinity) {
-		case corev1.ServiceAffinityNone:
-			service.Spec.SessionAffinity = corev1.ServiceAffinityNone
-		case corev1.ServiceAffinityClientIP:
-			service.Spec.SessionAffinity = corev1.ServiceAffinityClientIP
-		default:
-			return nil, fmt.Errorf("unknown session affinity: %s", o.SessionAffinity)
-		}
-	}
-	if len(o.ClusterIP) != 0 {
-		if o.ClusterIP == "None" {
-			service.Spec.ClusterIP = corev1.ClusterIPNone
-		} else {
-			service.Spec.ClusterIP = o.ClusterIP
-		}
-	}
-	return &service, nil
-}
-
-// parseLabels turns a string representation of a label set into a map[string]string
-func parseLabels(labelSpec string) (map[string]string, error) {
-	if len(labelSpec) == 0 {
-		return nil, fmt.Errorf("no label spec passed")
-	}
-	labels := map[string]string{}
-	labelSpecs := strings.Split(labelSpec, ",")
-	for ix := range labelSpecs {
-		labelSpec := strings.Split(labelSpecs[ix], "=")
-		if len(labelSpec) != 2 {
-			return nil, fmt.Errorf("unexpected label spec: %s", labelSpecs[ix])
-		}
-		if len(labelSpec[0]) == 0 {
-			return nil, fmt.Errorf("unexpected empty label key")
-		}
-		labels[labelSpec[0]] = labelSpec[1]
-	}
-	return labels, nil
-}
-
-// parseProtocols turns a string representation of a protocols set into a map[string]string
-func parseProtocols(protocols string) (map[string]string, error) {
-	if len(protocols) == 0 {
-		return nil, fmt.Errorf("no protocols passed")
-	}
-	portProtocolMap := map[string]string{}
-	protocolsSlice := strings.Split(protocols, ",")
-	for ix := range protocolsSlice {
-		portProtocol := strings.Split(protocolsSlice[ix], "/")
-		if len(portProtocol) != 2 {
-			return nil, fmt.Errorf("unexpected port protocol mapping: %s", protocolsSlice[ix])
-		}
-		if len(portProtocol[0]) == 0 {
-			return nil, fmt.Errorf("unexpected empty port")
-		}
-		if len(portProtocol[1]) == 0 {
-			return nil, fmt.Errorf("unexpected empty protocol")
-		}
-		portProtocolMap[portProtocol[0]] = portProtocol[1]
-	}
-	return portProtocolMap, nil
+	return obj, nil
```

Comment:
What is the reason of these changes with respect to the referenced issue?

Reply:
Repetitive code implementation, which I extract as a public method

Reply:
Is this something required for fixing the bug or is this refactoring?.